File: CS/MvcQueryBuilder/Controllers/QueryBuilderController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace MvcQueryBuilder.Controllers {
    // Processes Query Builder requests.
    public class QueryBuilderController : DevExpress.Web.Mvc.Controllers.QueryBuilderApiControllerBase {
        public override ActionResult Invoke() {
            return base.Invoke();
        }

        public ActionResult Save() {
            var result = QueryBuilderExtension.GetSaveCallbackResult("QueryBuilder");
            Session["SelectCommand"] = result.SelectStatement;
            return RedirectToAction("Index", "Home");
        }
    }
}

File: VB/MvcQueryBuilder/Controllers/QueryBuilderController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace MvcQueryBuilder.Controllers

    ' Processes Query Builder requests.
    Public Class QueryBuilderController
        Inherits DevExpress.Web.Mvc.Controllers.QueryBuilderApiControllerBase

        Public Overrides Function Invoke() As ActionResult
            Return MyBase.Invoke()
        End Function

        Public Function Save() As ActionResult
            Dim result = QueryBuilderExtension.GetSaveCallbackResult("QueryBuilder")
            Session("SelectCommand") = result.SelectStatement
            Return RedirectToAction("Index", "Home")
        End Function
    End Class
End Namespace
